BioCollect/src/GALogin.m

```diff
@@ -60,58 +92,60 @@ - (IBAction)onClickRegister:(id)sender{
-    NSString *userName = self.usernameTextField.text;
-    NSString *password = self.passwordTextField.text;
-    [MRProgressOverlayView showOverlayAddedTo:appDelegate.window title:@"Processing.." mode:MRProgressOverlayViewModeIndeterminateSmall animated:YES];
-
-    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        NSError *error = nil;
-        [appDelegate.restCall  authenticate:userName password:password error:&error];
-        dispatch_async(dispatch_get_main_queue(), ^{
-            //Dismiss the ui indicator modal
-            [MRProgressOverlayView dismissOverlayForView:appDelegate.window animated:YES];
-            
-            // Invalid user name and password
-            if(error != nil){
-                DebugLog(@"%@",[error localizedDescription]);
-                NSString *message = [error localizedDescription];
-                UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Error"
-                                                                message:message
-                                                               delegate:self
-                                                      cancelButtonTitle:@"Dismiss"
-                                                      otherButtonTitles:nil];
-                [alert show];
-            }
-            else{
-                self.passwordTextField.text = @"";
-                self.usernameTextField.text = @"";
-                //Dismiss the login modal
-                [appDelegate.window.rootViewController dismissViewControllerAnimated:YES completion:nil];
-                NSString *appType = [[NSBundle mainBundle] objectForInfoDictionaryKey: @"Bio_AppType"];
-                [UIView transitionWithView:appDelegate.window
-                                  duration:0.5
-                                   options:UIViewAnimationOptionTransitionFlipFromLeft
-                                animations:^{ appDelegate.window.rootViewController = appDelegate.ozHomeNC; }
-                                completion:nil];
-            }
-        });
-    });
-}
-
-#pragma mark - Text field delegate handler.
-
-- (BOOL)textFieldShouldReturn:(UITextField *)textField {
-    if(textField == self.usernameTextField){
-        [textField resignFirstResponder];
-        [self.passwordTextField becomeFirstResponder];
-    }
-    if(textField == self.passwordTextField){
-        [textField resignFirstResponder];
-        [self authenticate];
+    // Retrieve the OpenID configuration
+    OIDServiceConfiguration *configuration = [GASettings getOpenIDConfig];
+    NSString *bundleId = [[[NSBundle mainBundle] bundleIdentifier] stringByReplacingOccurrencesOfString:@".testing" withString:@""];
+    NSArray *bundleParts = [bundleId componentsSeparatedByString:@"."];
+    NSString *bundleName = [bundleParts lastObject];
+    
+    // Fix bundleName for bilbyblitz app
+    if ([bundleName isEqualToString:@"tracker"]) {
```

Comment:
curious to know why bundle name is changed.

Reply:
The app's bundle name on Android is `bilbyblitz`, so I added this for consistency (& so we don't need an extra redirect URL in the CAS/Cognito settings)